File: infracloud-inspection-mvp/apps/workflow/app/logger.py
from __future__ import annotations

import json
from datetime import datetime, timezone
from typing import Any

from .config import settings
from .redaction import redact_sensitive_data


def _serialize(value: Any) -> Any:
    if isinstance(value, Exception):
        return {
            "error_type": value.__class__.__name__,
            "error_message": str(value),
        }
    return value


def log(level: str, message: str, **fields: Any) -> None:
    payload = {
        "timestamp": datetime.now(timezone.utc).isoformat(),
        "level": level,
        "service": "infracloud-langgraph-workflow",
        "environment": settings.app_env,
        "message": message,
    }
    payload.update(
        redact_sensitive_data(
            {key: _serialize(value) for key, value in fields.items() if value is not None}
        )
    )
    print(json.dumps(payload, ensure_ascii=False))


class WorkflowLogger:
    def info(self, message: str, **fields: Any) -> None:
        log("info", message, **fields)

    def warn(self, message: str, **fields: Any) -> None:
        log("warn", message, **fields)

    def error(self, message: str, **fields: Any) -> None:
        log("error", message, **fields)


logger = WorkflowLogger()

File: infracloud-inspection-mvp/apps/api/lib/logger.js
const ENVIRONMENT = process.env.NODE_ENV || "development";
import { redactSensitiveData } from "./redact.js";

function normalizeFields(fields = {}) {
  const normalized = { ...fields };

  if (normalized.requestId && !normalized.request_id) {
    normalized.request_id = normalized.requestId;
    delete normalized.requestId;
  }

  if (normalized.actorId && !normalized.actor_id) {
    normalized.actor_id = normalized.actorId;
    delete normalized.actorId;
  }

  if (normalized.workflowRunId && !normalized.workflow_run_id) {
    normalized.workflow_run_id = normalized.workflowRunId;
    delete normalized.workflowRunId;
  }

  if (normalized.suspicionId && !normalized.suspicion_id) {
    normalized.suspicion_id = normalized.suspicionId;
    delete normalized.suspicionId;
  }

  const error = normalized.error;
  if (error instanceof Error) {
    normalized.error_type = error.name;
    normalized.error_message = error.message;
    delete normalized.error;
  }

  return normalized;
}

function write(level, message, fields = {}) {
  const payload = {
    timestamp: new Date().toISOString(),
    level,
    service: "infracloud-review-api",
    environment: ENVIRONMENT,
    message,
    ...redactSensitiveData(normalizeFields(fields)),
  };

  const line = JSON.stringify(payload);
  if (level === "error") {
    console.error(line);
    return;
  }

  console.log(line);
}

export const logger = {
  info(message, fields) {
    write("info", message, fields);
  },
  warn(message, fields) {
    write("warn", message, fields);
  },
  error(message, fields) {
    write("error", message, fields);
  },
};
